server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgModel.java

```diff
@@ -99,6 +99,15 @@ public boolean hasValue(int valuesAvailable) {
      */
     protected abstract double[] doPredict(Collection<Double> values, int numPredictions);
 
+    /**
+     * This method allows models to validate the window size if required
+     */
+    protected void validate(long window) {
```

Comment:
Should we pass the agg name into this method so we can add it to the exceptions it throws?

Reply:
:+1: